pkg/kafka/auth/config.go

```diff
@@ -51,14 +51,18 @@ func (config *AuthenticationConfig) SetConfiguration(saramaConfig *sarama.Config
 	if strings.Trim(authentication, " ") == "" {
 		authentication = none
 	}
+	if config.Authentication == tls || config.TLS.Enabled {
+		err := setTLSConfiguration(&config.TLS, saramaConfig)
+		if err != nil {
+			return err
+		}
+	}
 	switch authentication {
 	case none:
 		return nil
 	case kerberos:
 		setKerberosConfiguration(&config.Kerberos, saramaConfig)
 		return nil
-	case tls:
-		return setTLSConfiguration(&config.TLS, saramaConfig)
```

Comment:
if this case is removed, wouldn't `Authentication == tls` cause this to fail in the `default:` case?

Reply:
you are  right I have added it back